packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -3559,7 +3568,11 @@ function doubleInvokeEffectsInDEVIfNecessary(
     if (fiber.flags & PlacementDEV) {
       setCurrentDebugFiberInDEV(fiber);
       if (isInStrictMode) {
-        doubleInvokeEffectsOnFiber(root, fiber);
+        doubleInvokeEffectsOnFiber(
+          root,
+          fiber,
+          (fiber.mode & NoStrictPassiveEffectsMode) === NoMode,
```

Comment:
should the mode be passed down below on L3597 as well?

Reply:
Good question. I checked the test case where a Suspense boundary is used
https://github.com/facebook/react/blob/546fe4681c52de6a333a55cedb141c87b626425e/packages/react/src/__tests__/ReactStrictMode-test.internal.js#L130-L150

 and the codepath on L3597 doesn't hit. So no, we don't need this flag for components under a real Offscreen component.